apis/r/R/TileDBArray.R

```diff
@@ -9,6 +9,35 @@ TileDBArray <- R6::R6Class(
   inherit = TileDBObject,
   public = list(
 
+    #' @description Open the SOMA object for read or write.
+    #' @param internal_use_only Character value to signal 'permitted' call as
+    #' `new()` is considered internal and should not be called directly
+    #' @return The object, invisibly
+    open = function(mode="READ", internal_use_only = NULL) {
+      mode <- match.arg(mode, c("READ", "WRITE"))
```

Comment:
(Maybe we have to do it here because an R6-embedded function scopes differently but the nice things is that in a general R function we have the powerful (but harder to read the few times you see it) scheme of

```r
myfunc <- function(arg = c("READ", "WRITE")) { 
     arg <- match.arg(arg)      # by design this will get the values to compare against from the signatuer
     cat("All good\n")
}
```

which behave like this

```r
> myfunc <- function(arg = c("READ", "WRITE")) { 
     arg <- match.arg(arg)      # by design this will get the values to compare against from the signatuer
     cat("All good\n")
}
> 
> myfunc("READ")
All good
> myfunc("frob")
Error in match.arg(arg) : 'arg' should be one of “READ”, “WRITE”
> myfunc()       # there is a default argument: the first one in the vector
All good
> myfunc(NULL)   # idem, I guess. weirder
All good
> myfunc(NA)       # as expected
Error in match.arg(arg) : 'arg' must be NULL or a character vector
> 
```

Reply:
It is no different from what you have now.  `open = function(mode="READ", internal_use_only = NULL) `

That implies `READ` as the default value, assigned as fallback by default when no value is supplied in the call.  When you switch to `open = function(mode=c"READ","WRITE"), internal_use_only = NULL)` it is exactly the same but a) the signature is slightly longer which may matter but b) `mode <- match.arg(mode)` works and you win one round of DRY because you don't need the same value in two places.  Took me a while to come around to, now I find it more idiomatic.  Anyway....

Reply:
💥